test/e2e/packetcapture_test.go

```diff
@@ -670,3 +709,67 @@ func conditionSliceEqualsIgnoreLastTransitionTime(as, bs []crdv1alpha1.PacketCap
 	}
 	return true
 }
+
+// verifyPacketFile will read the packets file and check if packet count and packet data match with CR.
+func verifyPacketFile(t *testing.T, pc *crdv1alpha1.PacketCapture, reader io.Reader, targetNum int32, srcIP net.IP, dstIP net.IP) (err error) {
+	ngReader, err := pcapgo.NewNgReader(reader, pcapgo.DefaultNgReaderOptions)
+	if err != nil {
+		return err
+	}
+
+	for i := int32(0); i < targetNum; i++ {
+		data, _, err := ngReader.ReadPacketData()
+		if err != nil {
+			return err
+		}
+		packet := gopacket.NewPacket(data, layers.LayerTypeEthernet, gopacket.Default)
+		ipLayer := packet.Layer(layers.LayerTypeIPv4)
+		assert.NotNil(t, ipLayer)
```

Comment:
should be `require.NotNil`
If the assertion is false and you keep going, there will be a panic

I think the same comment applies to the transport protocol assertions below